src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -527,43 +528,18 @@ private static bool IsValidTokenChar(char c)
                 c == '~';
         }
 
-        public static void RejectRequest(RequestRejectionReason reason)
-        {
-            throw BadHttpRequestException.GetException(reason);
-        }
-
-        public static void RejectRequest(RequestRejectionReason reason, string value)
-        {
-            throw BadHttpRequestException.GetException(reason, value);
-        }
-
         private void RejectRequestLine(Span<byte> span)
         {
-            throw GetRejectRequestLineException(span);
-        }
-
-        private BadHttpRequestException GetRejectRequestLineException(Span<byte> span)
-        {
-            const int MaxRequestLineError = 32;
-            return BadHttpRequestException.GetException(RequestRejectionReason.InvalidRequestLine,
-                Log.IsEnabled(LogLevel.Information) ? span.GetAsciiStringEscaped(MaxRequestLineError) : string.Empty);
+            RequestRejectionUtilities.RejectRequest(
+                RequestRejectionReason.InvalidRequestLine,
+                (Log.IsEnabled(LogLevel.Information) ? span.GetAsciiStringEscaped(32) : string.Empty));
         }
 
         private unsafe void RejectRequestHeader(byte* headerLine, int length)
         {
-            RejectRequestHeader(new Span<byte>(headerLine, length));
-        }
-
-        private void RejectRequestHeader(Span<byte> span)
-        {
-            throw GetRejectRequestHeaderException(span);
-        }
-
-        private BadHttpRequestException GetRejectRequestHeaderException(Span<byte> span)
-        {
-            const int MaxRequestHeaderError = 128;
-            return BadHttpRequestException.GetException(RequestRejectionReason.InvalidRequestHeader,
-                Log.IsEnabled(LogLevel.Information) ? span.GetAsciiStringEscaped(MaxRequestHeaderError) : string.Empty);
+            RequestRejectionUtilities.RejectRequest(
```

Comment:
e.g. creating span, checking if logging is enabled, passing maxChars 128 etc can all be moved into second function that returns exception and takes the same params e.g. `GetRejectRequestHeaderExcepetion(byte* headerLine, int length)`

Then this could become
```csharp
private unsafe void RejectRequestHeader(byte* headerLine, int length)
{
    throw GetRejectRequestHeaderExcepetion(headerLine, length);
}
```